ch02.adoc

```diff
@@ -268,3 +268,216 @@ They may not be attached to a group, even if all variables within that group use
+Every fragment must have at least one version, but not all fragments need have the same number of versions.
+Where fragments have fewer versions than others, the trailing dimension must be padded with missing values.
+See <<example-L.2>>.
+
+*format*
+
+The string-valued `format` fragment array variable defines the format of the fragment files.
+In general it must have the same dimensions in the same order as the `file` fragment array variable, and must contain a non-missing value corresponding to each fragment version.
+However, if the `format` fragment array variable is a scalar, then its single value is assumed to apply to all fragments.
+The format of a netCDF fragment file must be indicated with the value `nc`.
+Other fragment file formats may be provided, on the understanding that an application program may choose to ignore any values that it does not understand.
+The `format` fragment array variable may contain a range of different values, i.e. not all fragment files need to have the same format, provided that the `address` fragment array variable can still be used to find each fragment within its fragment file. See <<example-L.1>>.
+
+*address*
+
+The `address` fragment array variable defines how to find each fragment within its fragment file, i.e. the address of the fragment.
+In general it must have the same dimensions in the same order as the `file` fragment array variable, and must contain a non-missing value corresponding to each fragment version.
+However, if the `address` fragment array variable is a scalar, then its single value is assumed to apply to all fragments.
+It may have any data type.
+For a netCDF fragment file, the string-valued address must be the fragment's netCDF variable name.
+Addresses for other fragment file formats are allowed, on the understanding that an application program may choose to ignore any values that it does not understand.
+See <<example-L.1>> and <<example-L.6>>.
+
+*shape*
+
+The integer-valued `shape` fragment array variable defines the shape of the data of each fragment in its canonical form (see <<fragment-interpretation>>).
+In general, the `shape` fragment array variable is two-dimensional, with the size of the slower varying dimension (i.e. the number of rows) being the number of fragment array dimensions, and the size of the more rapidly varying dimension (i.e. the number of columns) being the size of the largest fragment array dimension.
+The rows correspond to the fragment array dimensions in the same order, and each row provides the sizes of the fragments along that dimension of the fragment array, padded with missing values if there are fewer fragments than the number of columns.
+The sum of non-missing values in a row must therefore equal the size of the corresponding aggregated dimension.
+When the aggregated data is scalar there are no aggregated dimensions, and the `shape` fragment array variable must be one-dimensional and contain only the value `1` (hence be size one).
```

Comment:
Hi @sadielbartholomew - not so sure now, as the new text doesn't definitively preclude there being missing values
```suggestion
When the aggregated data is scalar there are no aggregated dimensions, and the `shape` fragment array variable must be one-dimensional, of size one, and contain the value `1`.
```

Reply:
Ah, good point! I didn't realise that, though the text does make it clear (in fairness to me, most people won't be reading the whole new section and appendix dedicated to CFA all at once - there's a lot to take in!). In which case, please ignore this suggestion.